feat: keep deleting forward with cmd+backspace

chenmoonmo opened the pull request:
This PR adds the same handling to "command + backspace" as "backspace" to support smooth paragraph deletion.
The following video contains a comparison with the notion and obsidian operations.


https://user-images.githubusercontent.com/36295999/212618163-f9e46652-7bec-46a0-b450-d5199ed6214e.mp4

doodlewind reviewed (commented):
Thanks for your contribution! If I understand correctly, our current playground already supports `cmd-backspace` for single line deletion. The difference is that for custom behavior, e.g., pressing backspace at the start of an empty list, `cmd+backspace` doesn't "keep deleting", and IMO we only need to modify the `HOTKEYS` binding to make this happen (rather than adding quill binding). Also, would you mind adding a test case to verify this? Thanks.

chenmoonmo commented:
Yes, that's what I'm trying to say, I've modified the code as I understand it and added test cases, but I'm not sure if this meets your requirements.
BTW, this is the first test case I wrote in my life.

Himself65 commented:
rebase please?

doodlewind commented on packages/blocks/src/page-block/default/utils.ts:
```diff
@@ -337,7 +341,15 @@ export function bindHotkeys(
   hotkey.addListener(BACKSPACE, e => {
     const { state } = selection;
-    if (state.type === 'native') {
+    const target = e.target as HTMLElement;
```
This doesn't feel right since it adds an `isCollapsedAtBlockStart` before the simple conditions based on `state.type`. 🤔 Try merging the latest refactored branch and you'll see the inconsistence.

chenmoonmo commented:
I feel screwed up.

chenmoonmo commented:
I feel screwed up.

Himself65 commented:
？

Himself65 commented:
I think you git histry crashed, let me handle this

Himself65 commented:
I'm also feel lost, don't know what the code should be...

Himself65 commented:
I think now the code is clean @chenmoonmo
does this code make sense to you?

lawvs commented on packages/blocks/src/__internal__/utils/consts.ts:
```diff
@@ -6,7 +6,7 @@ export const ALLOW_DEFAULT = true;
   UNDO: 'command+z,ctrl+z',
   REDO: 'command+shift+z,ctrl+shift+z,ctrl+y',
-  BACKSPACE: 'backspace',
+  BACKSPACE: 'backspace,command+backspace',
```
Is `crtl+backspace` should be added?

chenmoonmo commented:
I tried to merge and rebase the branches @Himself65, sorry for my mistakes

lawvs commented:
I refactor the code and now it is ready to merge.

Himself65 commented on tests/slash-menu.spec.ts:
```diff
@@ -38,9 +37,17 @@ test.describe('slash menu should show and hide correctly', () => {
+      const { page } = window;
+      const block = page.getBlockById(paragraphId);
+      if (!block) {
+        throw new Error('Block not found, id: ' + paragraphId);
```
use `assertExists`

lawvs replied:
This is in the test file

doodlewind commented:
Thanks @chenmoonmo for the enhancement! BTW we are also hosting Chinese community chat in https://t.me/blocksuite_cn, welcome!
